src/common/collections.rs

```diff
@@ -50,6 +50,13 @@ pub async fn do_list_collections(toc: &TableOfContent) -> CollectionsResponse {
     CollectionsResponse { collections }
 }
 
+pub async fn do_check_collection_exists(
+    toc: &TableOfContent,
+    name: &str,
+) -> Result<bool, StorageError> {
+    Ok(toc.get_collection(name).await.is_ok())
```

Comment:
here if any errors happen inside `get_collection`, we return `false` which is not the desired result. it's better to expose the error.

Reply:
Sure, I'll get this fixed